File: WeeklyCode/week2/Reverse Linked List II/chenhao.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
    ListNode *reverseBetween(ListNode *head, int m, int n) {
        ListNode vHead=ListNode(0);
        vHead.next=head;

        ListNode *cur=head, *pre=&vHead;
        ListNode *preBegin, *begin;

        for(int i=1;cur;i++){
            ListNode* next=cur->next;
            if(i==m){
                preBegin=pre;
                begin=cur;
            }
            if(i>m){
                cur->next=pre;    
            }
            if(i==n){
                preBegin->next=cur;
                begin->next=next;
                break;
            }
            pre=cur;
            cur=next;
        }
        return vHead.next;
    }
};

File: WeeklyCode/week8/Jump Game II/chenhao.java
public class Solution {
    public int jump(int[] A) {
        if(A==null || A.length<=1){
            return 0;
        }
        int cur=0,next=0;
        int jumps=0;
        for(int i=0;i<A.length;){
            if(i>cur){
                cur=next;
                jumps++;
            }
            else{
                next=Math.max(next,i+A[i]);
                if(next>=A.length-1){
                    return jumps+1;
                }
                i++;
            }
        }
        return -1;
    }
}